### ISSUE TITLE
Laravel Horizon pop params wrong

### ISSUE THREAD

**Issue by bbashy** [laravel/framework#53063]
### Laravel Version

11.27.0

### PHP Version

8.3.12

### Description

After upgrading from 11.26.0 to 11.27.0 this error happens for `php artisan horizon`

```
 INFO  Shutting down.


   Symfony\Component\ErrorHandler\Error\FatalError

  Declaration of Laravel\Horizon\RedisQueue::pop($queue = null) must be compatible with Illuminate\Queue\RedisQueue::pop($queue = null, $index = 0)

  at vendor/laravel/horizon/src/RedisQueue.php:131
    127▕      *
    128▕      * @param  string  $queue
    129▕      * @return \Illuminate\Contracts\Queue\Job|null
    130▕      */
  ➜ 131▕     public function pop($queue = null)
    132▕     {
    133▕         return tap(parent::pop($queue), function ($result) use ($queue) {
    134▕             if ($result) {
    135▕                 $this->event($this->getQueue($queue), new JobReserved($result->getReservedJob()));


   Whoops\Exception\ErrorException

  Declaration of Laravel\Horizon\RedisQueue::pop($queue = null) must be compatible with Illuminate\Queue\RedisQueue::pop($queue = null, $index = 0)

  at vendor/laravel/horizon/src/RedisQueue.php:131
    127▕      *
    128▕      * @param  string  $queue
    129▕      * @return \Illuminate\Contracts\Queue\Job|null
    130▕      */
  ➜ 131▕     public function pop($queue = null)
    132▕     {
    133▕         return tap(parent::pop($queue), function ($result) use ($queue) {
    134▕             if ($result) {
    135▕                 $this->event($this->getQueue($queue), new JobReserved($result->getReservedJob()));

      +1 vendor frames

  2   [internal]:0
      Whoops\Run::handleShutdown()
```

Downgrading fixes it.

### Steps To Reproduce

Run `php artisan horizon` even if not using redis for queues.

**Comment by aromka** [comment]
Getting the same after the latest update (11.27.0) - horizon fails with `Symfony\Component\ErrorHandler\Error\FatalError: Declaration of Laravel\Horizon\RedisQueue::pop($queue = null) must be compatible with Illuminate\Queue\RedisQueue::pop($queue = null, $index = 0)`

**Comment by webpatser** [comment]
Also with this update and the addition of the --json option. Horizon won't start properly since the --json option is present in the horizon worker.

https://github.com/laravel/horizon/pull/1505

**Comment by crynobone** [comment]
It should be fixed via  laravel/horizon#1504
